TTD/LiveRecorderApiSample/main.cpp

```diff
@@ -0,0 +1,190 @@
+#include <Windows.h>
+#include <wrl.h>
+
+#include <thread>
+#include <future>
+#include <filesystem>
+#include <format>
+#include <iostream>
+#include <random>
+#include <span>
+
+#include <TTD/TTDLiveRecorder.h>
+
+#pragma comment(lib, "TTDLiveRecorder")
+
+// {6DA58208-3BF5-4B80-A711-781098BC4445}
+static constexpr GUID ClientGuid = { 0x6da58208, 0x3bf5, 0x4b80, { 0xa7, 0x11, 0x78, 0x10, 0x98, 0xbc, 0x44, 0x45 } };
+
+// We define what the activity IDs mean.
+constexpr TTD::ActivityId StdSortActivity          { 1 };   // Call a std::sort function.
+constexpr TTD::ActivityId MultithreadedSortActivity{ 2 };   // A multithreaded sort.
+
+// We wish to report recording states by name.
+char const* GetStateName(TTD::ThreadRecordingState state)
+{
+    switch (state)
+    {
+    case TTD::ThreadRecordingState::NotStarted: return "NotStarted";
+    case TTD::ThreadRecordingState::Paused    : return "Paused";
+    case TTD::ThreadRecordingState::Recording : return "Recording";
+    case TTD::ThreadRecordingState::Throttled : return "Throttled";
+    default                                   : return "<unknown state>";
+    }
+}
+
+// Multithreaded sort function.
+// This algorithm splits the sorted range in two and runs one of the two pieces asynchronously.
+// It then merges the two pieces together, in-place.
+// it does this recursively: each of the two pieces is in turn split,
+// down to a reasonably (arbitrary) small size which is then sorted using std::sort.
+void MultithreadedSort(TTD::ILiveRecorder* pRecorder, std::span<int> span)
+{
+    if (span.size() <= 100)
+    {
+        std::ranges::sort(span);
+    }
+    else
+    {
+        auto const midPoint = span.size() / 2;
```

Comment:
Well... it only works with arithmetic types and pointers. If it worked with iterators, I'd be tempted to use it here instead of subspan. `std::midpoint(0, span.size())` doesn't really seem like an improvement here.